runtime/compiler/x/codegen/X86PrivateLinkage.cpp

```diff
@@ -1013,6 +1013,29 @@ void J9::X86::PrivateLinkage::createPrologue(TR::Instruction *cursor)
          }
       }
 
+   if (comp()->isDLT() && cg()->getSnippetList().empty())
+      {
+      TR::Instruction *lastInstruction = cg()->getAppendInstruction();
+
+      while (lastInstruction != NULL)
+         {
+         if (!lastInstruction->getOpCode().isPseudoOp())
+            {
+            if (lastInstruction->getOpCode().isCallOp())
+               {
+               // In some cases, the last instruction in the function will be a call instruction. The return
+               // address of the function may be beyond the body of this function. In this case, we must add
+               // padding to end of the function to prevent the stack walker from detecting an illegal return
+               // address.
+               generatePaddingInstruction(lastInstruction, 1, cg());
+               }
+            break;
+            }
+
+         lastInstruction = lastInstruction->getPrev();
+         }
+      }
```

Comment:
It is surprising and unintuitive to find this logic in `createPrologue()` as this has nothing to do with the prologue as far as I can tell.  From a purely conceptual integrity standpoint it doesn't belong here.  Can you explain the rationale or clarify your thinking for me please?

Ironically, this logic inserts an instruction in the farthest spot from the prologue as possible. :-)

Reply:
I figured it easiest to put it in the same place as is on [power](https://github.com/eclipse-openj9/openj9/blob/7f12be64b913204f8be35ec35bf55840e82a18f0/runtime/compiler/p/codegen/PPCPrivateLinkage.cpp#L1075-L1099). If you want, I'll move it into its own function.